elf/module.go

```diff
@@ -186,6 +186,24 @@ func writeKprobeEvent(probeType, eventName, funcName, maxactiveStr string) (int,
 	return kprobeId, nil
 }
 
+func perfEventOpenTracepoint(id int, progFd int) (int, error) {
+	efd, err := C.perf_event_open_tracepoint(C.int(id), -1 /* pid */, 0 /* cpu */, -1 /* group_fd */, C.PERF_FLAG_FD_CLOEXEC)
+	if efd < 0 {
+		return -1, fmt.Errorf("perf_event_open error: %v", err)
+	}
+
+	_, _, err2 := syscall.Syscall(syscall.SYS_IOCTL, uintptr(efd), C.PERF_EVENT_IOC_ENABLE, 0)
```

Comment:
I guess that doesn't work because you `syscall.Errno` can't be `nil`. We can do this though:

```diff
diff --git a/elf/module.go b/elf/module.go
index 5a6b940..9b850d3 100644
--- a/elf/module.go
+++ b/elf/module.go
@@ -202,14 +202,12 @@ func perfEventOpenTracepoint(id int, progFd int) (int, error) {
                return -1, fmt.Errorf("perf_event_open error: %v", err)
        }
 
-       _, _, err2 := syscall.Syscall(syscall.SYS_IOCTL, uintptr(efd), C.PERF_EVENT_IOC_ENABLE, 0)
-       if err2 != 0 {
-               return -1, fmt.Errorf("error enabling perf event: %v", err2)
+       if _, _, err := syscall.Syscall(syscall.SYS_IOCTL, uintptr(efd), C.PERF_EVENT_IOC_ENABLE, 0); err != 0 {
+               return -1, fmt.Errorf("error enabling perf event: %v", err)
        }
 
-       _, _, err2 = syscall.Syscall(syscall.SYS_IOCTL, uintptr(efd), C.PERF_EVENT_IOC_SET_BPF, uintptr(progFd))
-       if err2 != 0 {
-               return -1, fmt.Errorf("error attaching bpf program to perf event: %v", err2)
+       if _, _, err := syscall.Syscall(syscall.SYS_IOCTL, uintptr(efd), C.PERF_EVENT_IOC_SET_BPF, uintptr(progFd)); err != 0 {
+               return -1, fmt.Errorf("error attaching bpf program to perf event: %v", err)
        }
        return int(efd), nil
 }
```